### PR TITLE
fix node ranges and remove empty nodes

### PR DESCRIPTION
**Pull request by patricklx**
closes #35

### CODE REVIEW

**Comment by patricklx** [comment]
https://github.com/babel/babel/pull/16240 ...

**Comment by patricklx** [comment]
i think we can just merge this.
the current workaround for this issue would be to use overrides to set the version for `@babel/eslint-parser` to `^7.23.3`

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -37,7 +37,8 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - uses: wyvox/action-setup-pnpm@v3
-      - run: pnpm --filter '*' test 
+      - run: pnpm jest
```
Comment (NullVoxPopuli) [inline .github/workflows/ci.yml:40]: Can we use vitest instead of jest?

Jest has been a royal pain, and i don't want to give them more usage:/

Comment (patricklx) [reply .github/workflows/ci.yml:40]: sure, I like vite

File: __tests__/parser-test.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    const nodes = [];
+    traverse(result.visitorKeys, result.ast, (path) => {
+      nodes.push(path.node);
+    });
+    expect(nodes.map(n => [n.type, text.slice(...n.range)])).toMatchSnapshot();
+  });
+
+  it('token ranges are correct', () => {
+    const tokens = result.ast.tokens;
+    expect(tokens.map(n => [n.type, text.slice(...n.range)])).toMatchSnapshot();
```
Comment (NullVoxPopuli) [inline __tests__/parser-test.js:77]: Why snapshots instead of inline snapshots?

Comment (patricklx) [reply __tests__/parser-test.js:77]: doesnt support prettier v3... :(

File: package.json [hunk 3]
```diff
@@ -19,13 +19,14 @@
   ],
   "scripts": {
     "lint": "concurrently \"npm:lint:*(!fix)\" --names \"lint:\"",
-    "lint:js": "eslint .",
```
Comment (NullVoxPopuli) [inline package.json:22]: We still want to lint!

**Review by NullVoxPopuli — APPROVED** [review]
Just some tooling requests, otherwise everything looks good. Thank you!!

File: __tests__/parser-test.js (hunk 2 above)
Comment (NullVoxPopuli) [reply __tests__/parser-test.js:77]: Works fine in vitest

File: package.json (hunk 3 above)
Comment (patricklx) [reply package.json:22]: oops, will revert

**Comment by patricklx** [comment]
@NullVoxPopuli done

File: tests/parser.test.js [hunk 4]
```diff
@@ -0,0 +1,5288 @@
...
+
+  it('all tokens are correct', () => {
+    expect(
+      result.ast.tokens.map((t) => ({
+        value: t.value,
+        loc: t.loc,
+        range: t.range,
+      }))
+    ).toMatchInlineSnapshot(`
+      [
```
Comment (NullVoxPopuli) [inline tests/parser.test.js:60]: Is there a way we can reduce this?

For asserting an internal structure, it seems sike something this big would be hard to work with/debug?

File: package.json (hunk 3 above)
Comment (NullVoxPopuli) [reply package.json:22]: Thanks!

File: tests/parser.test.js (hunk 4 above)
Comment (patricklx) [reply tests/parser.test.js]: tried some improvements

**Pull request merged**